server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/projection/cassandra/CassandraMailQueueProjectionConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.projection.cassandra;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.lang3.time.DateUtils;
+
+public class CassandraMailQueueProjectionConfiguration {
+
+    private final AtomicInteger bucketCount;
+
+    public CassandraMailQueueProjectionConfiguration(AtomicInteger bucketCount) {
+        this.bucketCount = bucketCount;
+    }
+
+    public int increaseBucket() {
+        return bucketCount.incrementAndGet();
+    }
+
+    public AtomicInteger getBucketCount() {
+        return bucketCount;
+    }
+
+    public Date round(Date dateToRound) {
+        return DateUtils.round(dateToRound, Calendar.HOUR);
```

Comment:
just for temporally, i don't find a convenience way to do it. Let it for the next implementation